openapi-spec/swagger.yaml

```diff
@@ -2424,44 +2486,85 @@ components:
                 lvPath: string
                 nfsFileshareID: string
                 nfsFileshareName: string
-    FileShareSnapshotSpec:
-      description: Snapshot is a description of fileshare snapshot resource.
-      allOf:
-        - $ref: '#/components/schemas/BaseModel'
-        - type: object
-          required:
-            - tenantId
-            - name
-            - status
-            - snapshotId
-            - fileshareId
-          properties:
-            fileshareId:
-              type: string
-            tenantId:
-              type: string
-              readOnly: true
-            userId:
-              type: string
-              readOnly: true
-            name:
-              type: string
-            description:
-              type: string
-            status:
-              type: string
-              readOnly: true
-            profileId:
+    FileShareSnapshotCreateSpec:
+      description: FileShare Snapshot is a snapshot of file share created by nfs southbound driver, it can be
+        mounted on to physical machine or virtual machine instance.
+      - type: object
+        required:
+        - name
+        - fileshareId
+        properties:
+          fileshareId:
+            type: string
+          name:
+            minLength: 1
+            maxLength: 255
+            type: string
+          description:
+            type: string
+          profileId:
+            type: string
+          poolId:
+            type: string
+    FileShareSnapshotUpdateSpec:
+      description: Snapshot name and description can be updated using this specification.
+      - type: object
+        required:
+        - snapId
+        properties:
+          snapId:
+            type: string
```

Comment:
snapshotId is not required in snapshot updating body, because it's in the url path